File: Testing/tests/client/Tests/NotCompiling.cs
using Bridge;

namespace ClientTestLibrary
{
    //Does not compile issue [#59]
    [FileName("notCompiling.js")]
    class Class59
    {
        public static void Method1() { }

        public void Method1(Aux1 d) { }

        public class Aux1 { }
    }

    //Does not compile issue [#61]
    //TODO It shows a correct error description now. How to check that in tests?
    //[FileName("notCompiling.js")]
    //class Class61
    //{
    //    public static void Test1(bool condition, string message = null)
    //    {
    //        Bridge.Script.Write(string.Format("var q = {0};", 1));
    //    }
    //}

    //Error: Index was outside the bounds of the array.
    //public class Class2
    //{
    //    public static void Test1(bool condition, string message = null)
    //    {
    //        Bridge.Script.Write("");
    //    }
    //}


    //[#64]
    [FileName("notCompiling.js")]
    class Class64
    {
        public class Aux1 { }

        public Class64() { }

        public Class64(Aux1 related) { }

        public void Test()
        {
            var aux = new Class64.Aux1();
            new Class64(aux);
        }
    }

    //[#65]
    [FileName("notCompiling.js")]
    class Class65_1
    {
        public class Nested { }
    }

    [FileName("notCompiling.js")]
    class Class65_2 : Class65_1.Nested
    {
    }

    //[#66]
    [FileName("notCompiling.js")]
    public struct Rectangle66
    {
        public Rectangle66(int x1) { this = new Rectangle66(); }
        public Rectangle66(int x1, int x2) { }
    }

    [FileName("notCompiling.js")]
    //[#84] Does not compile
    class Class84
    {
        public void Test1()
        {
            try { }
            catch { }
        }
    }

    //[#89]
    //[FileName("notCompiling.js")]
    //public class Class89
    //{
    //    void Test(params object[] p)
    //    {
    //        var i = p[0];
    //    }
    //}

    //[#391]
    [FileName("notCompiling.js")]
    class Class391
    {
        public static void Main()
        {
            var TestArray1 = new[] { "TestA", "TestB", "TestC" };
            var TestArray2 = new[] { "TestA", "TestB", "TestC" };

            bool doSomething = false;
            foreach (var x in TestArray1)
            {
                foreach (var y in TestArray2)
                {
                    doSomething = x.Equals(y);
                }
            }

            foreach (var x in TestArray1)
            {
                foreach (var y in TestArray2)
                {
                    doSomething = x.Equals(y);
                }
            }
        }
    }
}

File: Testing/www/js/notCompiling.js
/* global Bridge */

"use strict";

Bridge.define('ClientTestLibrary.Class391', {
    statics: {
        main: function () {
            var $t, $t1, $t2, $t3;
            var TestArray1 = ["TestA", "TestB", "TestC"];
            var TestArray2 = ["TestA", "TestB", "TestC"];

            var doSomething = false;
            $t = Bridge.getEnumerator(TestArray1);
            while ($t.moveNext()) {
                var x = $t.getCurrent();
                $t1 = Bridge.getEnumerator(TestArray2);
                while ($t1.moveNext()) {
                    var y = $t1.getCurrent();
                    doSomething = Bridge.String.equals(x, y);
                }
            }

            $t2 = Bridge.getEnumerator(TestArray1);
            while ($t2.moveNext()) {
                var x1 = $t2.getCurrent();
                $t3 = Bridge.getEnumerator(TestArray2);
                while ($t3.moveNext()) {
                    var y1 = $t3.getCurrent();
                    doSomething = Bridge.String.equals(x1, y1);
                }
            }
        }
    }
});

Bridge.define('ClientTestLibrary.Class59', {
    statics: {
        method1: function () {
        }
    },
    method1: function (d) {
    }
});

Bridge.define('ClientTestLibrary.Class59.Aux1');

Bridge.define('ClientTestLibrary.Class64', {
    constructor: function () {
    },
    constructor$1: function (related) {
    },
    test: function () {
        var aux = new ClientTestLibrary.Class64.Aux1();
        new ClientTestLibrary.Class64("constructor$1", aux);
    }
});

Bridge.define('ClientTestLibrary.Class64.Aux1');

Bridge.define('ClientTestLibrary.Class65_1');

Bridge.define('ClientTestLibrary.Class65_1.Nested');

Bridge.define('ClientTestLibrary.Class65_2', {
    inherits: [ClientTestLibrary.Class65_1.Nested]
});

Bridge.define('ClientTestLibrary.Class84', {
    test1: function () {
        try {
        }
        catch ($e) {
            $e = Bridge.Exception.create($e);
        }
    }
});

Bridge.define('ClientTestLibrary.Rectangle66', {
    constructor$1: function (x1) {
        (new ClientTestLibrary.Rectangle66("constructor")).$clone(this);
    },
    constructor$2: function (x1, x2) {
    },
    constructor: function () {
    },
    $clone: function (to) { return this; }
});



Bridge.init();